plugins/source/aws/resources/services/timestream/databases.go

```diff
@@ -46,8 +46,14 @@ func Databases() *schema.Table {
 
 func fetchTimestreamDatabases(ctx context.Context, meta schema.ClientMeta, _ *schema.Resource, res chan<- any) error {
 	cl := meta.(*client.Client)
+	svc := cl.Services().Timestreamwrite
+	// This should be removed once https://github.com/aws/aws-sdk-go-v2/issues/2163 is fixed
+	if cl.AWSConfig != nil && cl.AWSConfig.Region != cl.Region {
+		cl.AWSConfig.Region = cl.Region
```

Comment:
Since `cl.AWSConfig` is a pointer (`*aws.Config`), couldn't we have a data race here when `fetchTimestreamDatabases` gets called by clients due to multiplexing? To be safe I would probably make a copy of `cl.AwsConfig` before changing the property and passing it in

Reply:
Good call. Updated accordingly